src/EnergyPlus/DataZoneEquipment.cc

```diff
@@ -1089,77 +1083,64 @@ int FindControlledZoneIndexFromSystemNodeNumberForZone(EnergyPlusData &state,
     // FUNCTION INFORMATION:
     //       AUTHOR         Brent Griffith
     //       DATE WRITTEN   August 2013
-    //       MODIFIED       na
-    //       RE-ENGINEERED  na
 
     // PURPOSE OF THIS FUNCTION:
     // This function returns the zone number for the indicated
     // zone node num.  Returns 0 if did not find zone node in any Zone
 
     // Return value
-    int ControlledZoneIndex; // Index into Controlled Zone structure
-
-    // FUNCTION LOCAL VARIABLE DECLARATIONS:
-    bool FoundIt;
-    int ZoneNum;
+    int ControlledZoneIndex = 0; // Index into Controlled Zone structure
 
-    FoundIt = false;
+    bool FoundIt = false;
 
     if (!state.dataZoneEquip->ZoneEquipInputsFilled) {
         GetZoneEquipmentData(state);
         state.dataZoneEquip->ZoneEquipInputsFilled = true;
     }
-    ControlledZoneIndex = 0;
-    for (ZoneNum = 1; ZoneNum <= state.dataGlobal->NumOfZones; ++ZoneNum) {
-        if (state.dataZoneEquip->ZoneEquipConfig(ZoneNum).ActualZoneNum > 0) {
+
+    for (int ZoneNum = 1; ZoneNum <= state.dataGlobal->NumOfZones; ++ZoneNum) {
+        if (state.dataZoneEquip->ZoneEquipConfig(ZoneNum).IsControlled) {
             if (TrialZoneNodeNum == state.dataZoneEquip->ZoneEquipConfig(ZoneNum).ZoneNode) {
                 // found it.
                 FoundIt = true;
-                ControlledZoneIndex = state.dataZoneEquip->ZoneEquipConfig(ZoneNum).ActualZoneNum;
+                ControlledZoneIndex = ZoneNum;
             }
         }
     }
 
     return ControlledZoneIndex;
 }
 
-int GetSystemNodeNumberForZone(EnergyPlusData &state, std::string const &ZoneName) // Zone name to match into Controlled Zone structure
+int GetSystemNodeNumberForZone(EnergyPlusData &state, int const zoneNum)
```

Comment:
Nice add on

Reply:
Every time a string is replaced with an integer or enum, five minutes is added to my life. Thank you!